ecs_client_test.go

```diff
@@ -19,11 +19,32 @@ import (
 func TestECSClientInterface(t *testing.T) {
 	assert.Implements(t, (*ECSClient)(nil), &BasicECSClient{})
 }
+func TestECSClientTaskDefinition(t *testing.T) {
+
+	cleanupTaskDefinition := func(ctx context.Context, t *testing.T, c *BasicECSClient, out *ecs.RegisterTaskDefinitionOutput) {
+		if out != nil && out.TaskDefinition != nil && out.TaskDefinition.TaskDefinitionArn != nil {
+			out, err := c.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{
+				TaskDefinition: out.TaskDefinition.TaskDefinitionArn,
+			})
+			require.NoError(t, err)
+			require.NotZero(t, out)
+		}
+	}
+
+	cleanupTask := func(ctx context.Context, t *testing.T, c *BasicECSClient, runOut *ecs.RunTaskOutput) {
+		out, err := c.StopTask(ctx, &ecs.StopTaskInput{
+			Cluster: aws.String(os.Getenv("AWS_ECS_CLUSTER")),
+			Task:    aws.String(*runOut.Tasks[0].TaskArn),
+		})
+		require.NoError(t, err)
+		require.NotZero(t, out)
+		require.NotZero(t, out.Task)
```

Comment:
Is there a particular reason you're checking the output on these and not on `cleanupTaskDefinition`?

If you're trying to test `RegisterTaskDefinition`/`DeregisterTaskDefinition`, I would write dedicated tests for those rather than cram the tests into cleanup helper functions. Cleanup code should _only_ be for cleanup, not testing behavior/functionality.

Reply:
Got it, fixed